File: Examples/VisualBasic/Articles/ManagingVBAModules/CheckVbaCodeIsSigned.vb
Imports Aspose.Cells
Imports Aspose.Cells.Vba
Imports Aspose.Cells.Drawing

Namespace Articles.ManagingVBAModules
    Public Class CheckVbaCodeIsSigned
        Public Shared Sub Run()
            ' ExStart:CheckVbaCodeIsSigned
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)

            Dim workbook As New Workbook(dataDir & Convert.ToString("sampleVBAProjectSigned.xlsm"))

            Console.WriteLine("Is VBA Code Project Signed: " + workbook.VbaProject.IsSigned)
            ' ExEnd:CheckVbaCodeIsSigned
        End Sub
    End Class
End Namespace

File: Examples/CSharp/Articles/ManagingVBAModules/CheckVbaCodeIsSigned.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Aspose.Cells.Examples.CSharp.Articles.ManagingVBAModules
{
    public class CheckVbaCodeIsSigned
    {
        public static void Run()
        {
            //Source directory
            string sourceDir = RunExamples.Get_SourceDirectory();

            Workbook workbook = new Workbook(sourceDir + "sampleCheckVbaCodeIsSigned.xlsm");

            Console.WriteLine("Is VBA Code Project Signed: " + workbook.VbaProject.IsSigned);

            Console.WriteLine("CheckVbaCodeIsSigned executed successfully.");
        }
    }
}
